server/plugin/api.go

```diff
@@ -84,6 +87,48 @@ func (p *Plugin) handleStaticFiles(r *mux.Router) {
 	r.PathPrefix("/static/").Handler(http.StripPrefix("/static/", http.FileServer(http.Dir(filepath.Join(bundlePath, "assets")))))
 }
 
+// handleFileAttachments returns the data of the fileID passed in the request URL.
+func (p *Plugin) handleFileAttachments(w http.ResponseWriter, r *http.Request) {
+	vars := mux.Vars(r)
+	encryptedFileInfo := vars[PathParamEncryptedFileInfo]
+
+	fileInfo := FileStruct{}
+
+	decoded, err := decode(encryptedFileInfo)
+	if err != nil {
+		p.API.LogError("Error occurred while decoding the file. Error: %s", err.Error())
+		return
+	}
+
+	jsonBytes, err := decrypt(decoded, []byte(p.getConfiguration().EncryptionSecret))
+	if err != nil {
+		p.API.LogError("Error occurred while decrypting the file. Error: %s", err.Error())
+		return
+	}
+
+	if err = json.Unmarshal(jsonBytes, &fileInfo); err != nil {
+		p.API.LogError("Error occurred while unmarshaling the file. Error: %s", err.Error())
+		return
+	}
+
+	if time.Now().After(fileInfo.Expiry) {
+		http.NotFound(w, r)
+		return
+	}
+
+	data, appErr := p.API.GetFile(fileInfo.ID)
+	if appErr != nil {
+		p.API.LogInfo("Couldn't get file data. FileID: %s", fileInfo.ID)
```

Comment:
Why are you not returning a response in any of the error handling conditions in this function?